taniumrest_connector.py

```diff
@@ -940,94 +988,119 @@ def _handle_get_question_results(self, param):
-                parameter_keys = [parameter["key"] for parameter in parameter_definition["parameters"]]
-                self.save_progress("Parameter Keys:\n{}".format(json.dumps(parameter_keys)))
-                total_params += len(parameter_keys)
-                parameters = []
-
-                for key in parameter_keys:
-                    if param_idx >= len(param_list):
-                        action_result.set_status(phantom.APP_ERROR, "For parameters which you do not want to add value, please use double quotes(\"\").\
-                                    For more details refer to the documentation")
-                        return
-
-                    parameter = {
-                        "key": "||%s||" % key,
-                        "value": param_list[param_idx] }
-                    parameters.append(parameter)
-                    param_idx += 1
-
-                sensor_dict = {
-                    "source_hash": sensor["hash"],
-                    "parameters": parameters }
-                sensor_data.append({"sensor": sensor_dict})
-            else:
-                # Regular Sensor, can use as-is
-                sensor_data.append({"sensor": sensor})
+        self.save_progress("Sensor Data:\n{}".format(json.dumps(sensor_data)))
 
-        if total_params and total_params != len(param_list):
+        if self._param_idx and self._param_idx != len(param_list):
             action_result.set_status(phantom.APP_ERROR, "Please provide the exact number of parameters expected by the sensor")
             return
 
-        self.save_progress("Sensor Data:\n{}".format(json.dumps(sensor_data)))
-        question_data = { "selects": sensor_data,
-                          "question_text": query["question_text"]}
-        if "group" in query:
-            # Add in filters
-            question_data["group"] = query["group"]
-
         return question_data
 
+    def _create_sensor_dict(self, action_result, sensor, param_list):
+        sensor_name = sensor["name"]
+        endpoint = TANIUMREST_GET_SENSOR_BY_NAME.format(sensor_name=sensor_name)
+        ret_val, response = self._make_rest_call_helper(action_result, endpoint, verify=self._verify)
+        if phantom.is_fail(ret_val):
+            action_result.set_status(phantom.APP_ERROR, "Failed to get sensor definition from Tanium")
+            return phantom.APP_ERROR, {}
+
+        response_data = response.get("data")
+        if not response_data:
+            action_result.set_status(phantom.APP_ERROR, TANIUMREST_RESOURCE_NOT_EXIST.format(sensor_name, "sensor"))
+            return phantom.APP_ERROR, {}
+
+        resp_data = self._get_response_data(response_data, action_result, "sensor")
+        if resp_data is None:
+            return phantom.APP_ERROR, {}
+
+        self.save_progress("Parameter Definition:\n{}".format(resp_data.get("parameter_definition", "")))
+
+        raw_parameter_definition = resp_data.get("parameter_definition", "")
```

Comment:
If `raw_parameter_definition` is None, should we exit early?

Reply:
returned the original dictionary if parameter definition is None (After json.loads())